src/goto-programs/goto_contractor.cpp

```diff
@@ -0,0 +1,338 @@
+//
+// Created by Mohannad Aldughaim on 09/01/2022.
+//
+
+#include "goto_contractor.h"
+
+void goto_contractort::get_constraints(goto_functionst goto_functions)
+{
+  auto function = goto_functions.function_map.find("c:@F@main");
+  auto it = function->second.body.instructions.begin();
+  while(it != function->second.body.instructions.end())
+  {
+    if(it->is_assert())
+      constraint = create_constraint_from_expr2t(it->guard);
+
+    it++;
+  }
+}
+
+void goto_contractort::get_intervals(goto_functionst goto_functions)
+{
+  auto function = goto_functions.function_map.find("c:@F@main");
+  auto it = function->second.body.instructions.begin();
+
+  while(it != function->second.body.instructions.end())
+  {
+    if(it->is_assume())
+    {
+      parse_intervals(it->guard);
+    }
+
+    it++;
+  }
+}
+
+void goto_contractort::parse_intervals(irep_container<expr2t> expr)
+{
+  symbol2tc symbol;
+  long value;
+
+  expr = get_base_object(expr);
+
+  if(is_and2t(expr))
+  {
+    parse_intervals(to_and2t(expr).side_1);
+    parse_intervals(to_and2t(expr).side_2);
+    return;
+  }
+
+  if(!is_comp_expr(expr))
+    return;
+
+  std::shared_ptr<relation_data> rel;
+  rel = std::dynamic_pointer_cast<relation_data>(expr);
+
+  //side1 1 is a symbol or typecast to symbol
+  auto side1 = rel->side_1;
+  //side 2 is always a number.
+  auto side2 = rel->side_2;
+
+  auto obj = get_base_object(side1);
+  if(!is_symbol2t(obj))
+    return;
+
+  symbol = to_symbol2t(obj);
+  bool neg = false;
+  if(is_neg2t(side2))
+  {
+    neg = true;
+    side2 = to_neg2t(side2).value;
+  }
+  if(!is_constant_int2t(side2))
+    return;
+  if(is_constant_int2t(side2))
```

Comment:
At this point we know that side2 is a constant_int2tc, so no need to check it again.

Reply:
> Using BigInt should do the trick... When you need to convert to double just call .toint64(). 

May I ask you to clarify? i didnt understand what you mean by using BigInt.

Reply:
BigInt is a type that can hold large integers and that can be converted to `int64_t` and `uin64_t`. Since you don't seem to be handling fp intervals, you can make `value` a `BigInt`, then call `value.to_int64()` when calling the `ibex` functions.